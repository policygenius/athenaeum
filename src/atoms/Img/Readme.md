### Img Example:

The `<Img />` component can take also any prop that an HTML5 img can take.

```jsx
    <Img />
```
