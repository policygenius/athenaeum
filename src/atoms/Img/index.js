import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './img.module.scss';

const imgixSrcset = src => `
  ${src}?fit=crop&w=320w&auto=format 320w,
  ${src}?fit=crop&w=768w&auto=format 768w,
  ${src}?fit=crop&w=1025&auto=format 1025w
`;

function Img(props) {
  const {
    className,
    src,
    srcSet,
    imgixSrc,
    alt,
    ...rest
  } = props;

  const classes = [
    styles['img'],
    className,
  ];

  return (
    <img
      className={classnames(...classes)}
      alt={alt}
      src={src || `https://policygenius-images.imgix.net/${imgixSrc}?fit=crop&w=768&auto=format&q=40`}
      srcSet={srcSet || imgixSrcset(`https://policygenius-images.imgix.net/${imgixSrc}`)}
      {...rest}
    />
  );
}


Img.propTypes = {
  /**
   * This prop will add a new className to any inherent classNames
   * provided in the component's index.js file.
   */
  className: PropTypes.string,

  /**
   * Recommended: The Imgix base url for the image. Will handle all optomizations automagically.
   * https://docs.imgix.com/apis/url/adjustment/usmrad
   */
  imgixSrc: PropTypes.string,

  /**
   * The src url for the image. If you use this, it will override the Imgix src.
   */
  src: PropTypes.string,

  /**
   * The srcset for the image. If you use this, it will override the Imgix src.
   */
  srcSet: PropTypes.string,

  /**
   * The img alt for the image. Defaults to an empty string.
   */
  alt: PropTypes.string,
};

Img.defaultProps = {
  alt: '',
  imgixSrc: 'photos/alexander-dummer-150646.jpg'
};

export default Img;
