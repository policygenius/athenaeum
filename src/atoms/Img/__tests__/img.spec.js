import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Img from 'atoms/Img';

describe('<Img />', () => {
  it('renders correctly', () => {
    const wrapper = mount(<Img />);

    expect(wrapper.type()).toEqual(Img);
  });

  it('renders correctly', () => {
    const actual = renderer.create(<Img />).toJSON();

    expect(actual).toMatchSnapshot();
  });
});
