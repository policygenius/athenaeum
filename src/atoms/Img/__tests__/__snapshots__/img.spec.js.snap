// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Img /> renders correctly with default props 1`] = `
<img
  alt="alexander dummer 150646"
  className="img"
  data-sizes="100vw"
  data-src="https://policygenius-images.imgix.net/photos/alexander-dummer-150646.jpg?fit=max&auto=format&ch=Width,DPR&w={width}"
  data-srcset="
  https://policygenius-images.imgix.net/photos/alexander-dummer-150646.jpg?fit=max&auto=format&ch=Width,DPR&w={width}&dpr=1 1x,
  https://policygenius-images.imgix.net/photos/alexander-dummer-150646.jpg?fit=max&auto=format&ch=Width,DPR&w={width}&dpr=2 2x,
  https://policygenius-images.imgix.net/photos/alexander-dummer-150646.jpg?fit=max&auto=format&ch=Width,DPR&w={width}&dpr=3 3x,
"
  src="https://policygenius-images.imgix.net/photos/alexander-dummer-150646.jpg?fit=max&auto=format&ch=Width,DPR&w={width}"
  title="alexander dummer 150646"
/>
`;

exports[`<Img /> renders correctly with given imgix picture source props 1`] = `
<picture
  className="picture"
>
  <source
    media="(max-width: 767px)"
    srcSet="https://policygenius-images.imgix.net/small?fit=max&auto=format&ch=Width,DPR&w={width}"
  />
  <source
    media="(max-width: 1024px)"
    srcSet="https://policygenius-images.imgix.net/medium?fit=max&auto=format&ch=Width,DPR&w={width}"
  />
  <img
    alt="default"
    className="img"
    data-sizes="auto"
    data-src="https://policygenius-images.imgix.net/default.jpg?fit=max&auto=format&ch=Width,DPR&w={width}"
    desktopImgixSrc="large"
    src="https://policygenius-images.imgix.net/default.jpg?fit=max&auto=format&ch=Width,DPR&w={width}"
    title="default"
  />
</picture>
`;

exports[`<Img /> renders correctly with given imgix props 1`] = `
<img
  alt="are cool"
  className="img cool-class"
  data-sizes="100vw"
  data-src="https://policygenius-images.imgix.net/photos/are-cool.jpg?fit=max&auto=format&ch=Width,DPR&w={width}"
  data-srcset="
  https://policygenius-images.imgix.net/photos/are-cool.jpg?fit=max&auto=format&ch=Width,DPR&w={width}&dpr=1 1x,
  https://policygenius-images.imgix.net/photos/are-cool.jpg?fit=max&auto=format&ch=Width,DPR&w={width}&dpr=2 2x,
  https://policygenius-images.imgix.net/photos/are-cool.jpg?fit=max&auto=format&ch=Width,DPR&w={width}&dpr=3 3x,
"
  maxHeight="20"
  maxWidth="10"
  src="https://policygenius-images.imgix.net/photos/are-cool.jpg?fit=max&auto=format&ch=Width,DPR&w={width}"
  title="are cool"
/>
`;

exports[`<Img /> renders correctly with given regular picture source props 1`] = `
<picture
  className="picture"
>
  <source
    media="(max-width: 767px)"
    srcSet="small"
  />
  <source
    media="(max-width: 1024px)"
    srcSet="medium"
  />
  <img
    alt="Alternative"
    className="img"
    data-sizes="auto"
    data-src="default.jpg"
    desktopSrc="large"
    src="default.jpg"
    title="Alternative"
  />
</picture>
`;

exports[`<Img /> renders correctly with given regular props 1`] = `
<img
  alt="alt"
  className="img cool-class"
  data-sizes="100vw"
  data-src="some-url"
  data-srcset="some-url"
  maxHeight="20"
  maxWidth="10"
  src="some-url"
  title="title"
/>
`;

exports[`<Img /> renders with hide classes when null src props are provided 1`] = `
<picture
  className="picture hide-mobile-image hide-tablet-image"
>
  <img
    alt="default"
    className="img"
    data-sizes="auto"
    data-src="https://policygenius-images.imgix.net/default.jpg?fit=max&auto=format&ch=Width,DPR&w={width}"
    src="https://policygenius-images.imgix.net/default.jpg?fit=max&auto=format&ch=Width,DPR&w={width}"
    title="default"
  />
</picture>
`;
